From a84100dce8a32de89c6a854c965c0c54f1126563 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date: Mon, 2 Jan 2017 02:16:40 +0100
Subject: [PATCH 1/2] Make activities kioslave optional

Not everyone builds this for a Plasma desktop, Qt5Sql is already only
searched for in that subdirectory, so this gives some nice modularity.

REVIEW: 129783
---
 CMakeLists.txt | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index edc2639..5ac418b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,7 +39,6 @@ find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS
     Solid
     Bookmarks
     GuiAddons
-    Activities
 )
 # As this is the check used for linkage, only require it in the same location...
 if (UNIX)
@@ -56,6 +55,12 @@ include(ECMOptionalAddSubdirectory)
 
 add_definitions(-DQT_NO_URL_CAST_FROM_STRING)
 
+find_package(KF5Activities QUIET)
+set_package_properties(KF5Activities PROPERTIES
+    PURPOSE "Provides the activities:/ kioslave and fileitem plugin."
+    TYPE OPTIONAL
+)
+
 find_package(SLP)
 set_package_properties(SLP PROPERTIES DESCRIPTION "SLP (Service Location Protocol) implementation"
                        URL "http://www.openslp.org/"
@@ -126,7 +131,9 @@ endif()
 add_subdirectory( doc )
 
 add_subdirectory( about )
-add_subdirectory( activities )
+if(KF5Activities_FOUND)
+  add_subdirectory( activities )
+endif()
 add_subdirectory( bookmarks )
 add_subdirectory( filter )
 if(Phonon4Qt5_FOUND)
-- 
2.10.2

